src/prommis/solvent_extraction/sx_model_flowsheet_dynamic.py

```diff
@@ -0,0 +1,186 @@
+from idaes.core.initialization import InitializationStatus
+from idaes.core.initialization.block_triangularization import (
+    BlockTriangularizationInitializer,
+)
+
+from prommis.leaching.leach_solution_properties import LeachSolutionParameters
+from prommis.solvent_extraction.ree_og_distribution import REESolExOgParameters
+from prommis.solvent_extraction.solvent_extraction import SolventExtraction
+
+"""
+Method of building a dynamic solvent extraction model with a specified number of 
+stages and with two separate property packages for the two inlet streams.
+This is a loading operation, so no additional argument has to be specified.
+
+"""
+
+m = ConcreteModel()
+
+time_duration = 60
+
+m.fs = FlowsheetBlock(dynamic=True, time_set=[0, time_duration], time_units=units.hour)
+
+m.fs.prop_o = REESolExOgParameters()
+m.fs.leach_soln = LeachSolutionParameters()
+
+number_of_stages = 3
+
+m.fs.solex = SolventExtraction(
+    number_of_finite_elements=number_of_stages,
+    aqueous_stream={
+        "property_package": m.fs.leach_soln,
+        "flow_direction": FlowDirection.forward,
+        "has_energy_balance": False,
+        "has_pressure_balance": False,
+    },
+    organic_stream={
+        "property_package": m.fs.prop_o,
+        "flow_direction": FlowDirection.backward,
+        "has_energy_balance": False,
+        "has_pressure_balance": False,
+    },
+)
+
+"""
+Discretization of the time domain, and specification of the partition coefficients,
+volume, volume fractions, and the initial conditions of state variables for the components
+for all the stages. 
+
+"""
+
+m.discretizer = TransformationFactory("dae.finite_difference")
+m.discretizer.apply_to(m, nfe=5, wrt=m.fs.time, scheme="BACKWARD")
+
+"""
+Specifications of the partition coefficients, volume and volume fractions for all
+the stages.
+
+"""
+
+m.fs.solex.mscontactor.volume[:].fix(0.4)
+
+m.fs.solex.mscontactor.volume_frac_stream[:, :, "aqueous"].fix(0.5)
+
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Al"] = 3.6 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Ca"] = 3.7 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Fe"] = 2.1 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Sc"] = 99.9 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Y"] = 99.9 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "La"] = 75.2 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Ce"] = 95.7 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Pr"] = 96.5 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Nd"] = 99.2 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Sm"] = 99.9 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Gd"] = 98.6 / 100
+m.fs.solex.partition_coefficient[:, "aqueous", "organic", "Dy"] = 99.9 / 100
+
+"""
+Fixation of the inlet conditions and the initial state values for all the components.
```

Comment:
Typo: "Fixing inlet...", not "Fixation of the inlet ..."

Reply:
Okay

Reply:
Could you please fix this?